pkg/config/config.go

```diff
@@ -201,6 +202,11 @@ func load(generalFilePath, hostsFilePath string) (*Config, error) {
 		generalMap.AddEntry("hosts", hostsMap)
 	}
 
+	if generalMap.Empty() && fallback != nil {
+		//TODO: Do we need to copy fallback?
```

Comment:
Open question here that could use some input.

Reply:
Yeh it's kinda funky because config isn't just a bag of data, it's actually a mutable structure. It's possible for the caller to end up modifying the fallback later without understanding the consequences. I had thought about providing a string before but that kind of leaked our internal config representation which would make it harder to change in the future.

I could be persuaded either way. Either we defensive copy and document to the caller that we are taking the values from the config, or we document that this **is** the Config struct that will be used as the fallback.

Reply:
I am going to change it to make a copy. That feels like the safer move.